### PR TITLE
#162948798 Create routers

### PR DESCRIPTION
**Pull request by kwameasiago**
### What does this PR do?
Create two routes for  login and home page

### Description of tasks accomplished

- Installed and configured Routers
- Added two routes login and Home
- Created directories component and view respectively 
- Added dummy a component and view

### How this app should be tested?
clone this repo and checkout to ch-setup-react-router-162948798. install package requirement and run npm start to run the app

### Related pivotal tracker stories
[#162948798](https://www.pivotaltracker.com/story/show/162948798)

### Screenshots
screenshot 2019-02-07 at 14 27 48

screenshot 2019-02-07 at 14 27 56

### CODE REVIEW

File: src/components/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+class Login extends Component{
+    render(){
+        return(
+            <div>
+                <h1>This is the login page</h1>
+            </div>
+        )
+    }
+}
+export default Login;
```
Comment (Georgeygigz) [inline src/components/Login.jsx]: Add a blank line at the end this file

File: src/view/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+class Home extends Component{
+    render(){
+        return(
+            <div>
+                <h1>This the home page</h1>
+            </div>
+        )
+    }
+}
+export default Home;
```
Comment (Georgeygigz) [inline src/view/Home.jsx]: ditto

**Review by MaggieChege — APPROVED** [review]
Looks good to me

**Review by Georgeygigz — APPROVED** [review]
Looks good to me

File: src/App.jsx [hunk 3]
```diff
@@ -1,30 +1,21 @@
...
-            href="https://reactjs.org"
-            target="_blank"
-            rel="noopener noreferrer"
-          >
-            Learn React
-          </a>
-        </header>
-      </div>
+      <BrowserRouter>
+      {/* <div> */}
```
Comment (KMaina) [inline src/App.jsx]: Kindly remove this line of code that is commented

File: src/App.jsx [hunk 4]
```diff
@@ -1,30 +1,21 @@
...
+        <Switch>
+          <Route exact path='/' component={Home}></Route>
+          <Route path='/login' component={Login}></Route>
+        </Switch>
+       
+      {/* </div> */}
```
Comment (KMaina) [inline src/App.jsx]: ditto

**Review by KMaina — APPROVED** [review]

File: src/App.jsx [hunk 5]
```diff
@@ -1,30 +1,18 @@
...
-            className="App-link"
-            href="https://reactjs.org"
-            target="_blank"
-            rel="noopener noreferrer"
-          >
-            Learn React
-          </a>
-        </header>
-      </div>
+      <BrowserRouter>
```
Comment (cosmas28) [inline src/App.jsx]: It's a good practice to create a route component on it's on file. ie `src/routes/routes.js`. Therefore, move these dummy routes to their own directory and nest the `Route component` as a child component of the `App component`.

**Review by cosmas28 — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 5 above)
Comment (kwameasiago) [reply src/App.jsx]: Ok will do that

File: src/components/routes.jsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import React, { Component } from 'react';
+import { BrowserRouter, Route, Switch } from 'react-router-dom';
+import Home from '../view/Home';
+import Login from './Login';
+
+class Routes extends Component {
```
Comment (cosmas28) [inline src/components/routes.jsx]: Any reason why this is a class-based component?

**Review by cosmas28 — CHANGES_REQUESTED** [review]
I requested the `Route component` to be under `src/routes/routes.jsx` tree. Why did you move it to `src/components`?

File: src/components/routes.jsx (hunk 6 above)
Comment (kwameasiago) [reply src/components/routes.jsx]: No reason. Just preference, Should I change to functional components?

File: src/view/Home.jsx [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+
+const Home = () => {
+        return (
```
Comment (cosmas28) [inline src/view/Home.jsx]: Kindly fix the indentation from line 4 to line 8.

File: src/view/Home.jsx [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+            <div>
+                <h1>This the home page</h1>
+            </div>
+        );
+};
+export default Home;
```
Comment (cosmas28) [inline src/view/Home.jsx:11]: There should be a line space before this line. This makes your code more readable.

**Review by cosmas28 — CHANGES_REQUESTED** [review]

File: src/view/Home.jsx (hunk 8 above)
Comment (cosmas28) [reply src/view/Home.jsx:11]: The indentation space should be consistent in all components. Kindly use two spaces like the way it is on `Login` and `Routes` components.

**Review by cosmas28 — CHANGES_REQUESTED** [review]

**Review by cosmas28 — APPROVED** [review]

**Pull request merged**